pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -403,6 +403,9 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 			status.ReadyReplicas++
 			if getPodRevision(pods[i]) == updateRevision.Name {
 				status.UpdatedReadyReplicas++
+				if avail, _ := isRunningAndAvailable(pods[i], minReadySeconds); avail {
+					status.UpdatedAvailableReplicas++
+				}
```

Comment:
[File](https://github.com/openkruise/kruise/blob/master/pkg/controller/statefulset/stateful_set_control_test.go) have many checks for **set.Status.UpdatedReplicas**, and i think you can add UpdatedAvailableReplicas checks.